addons/account/models/account_move.py

```diff
@@ -1302,6 +1303,69 @@ def _compute_quick_edit_mode(self):
             else:
                 move.quick_edit_mode = False
 
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    def _compute_duplicated_ref_ids(self):
+        move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
+        for move in self:
+            move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
+
+    def _fetch_duplicate_supplier_reference(self, only_posted=False):
+        moves = self.filtered(lambda m: m.is_purchase_document() and m.ref)
+        if not moves:
+            return {}
+
+        # this case deals with single move and thus avoid issue with those edited by UI
+        if len(moves) == 1:
```

Comment:
In which case is a `if not moves.ids` not working? Removing the need for the `if moves.ids else` down below?